doc/classes/String.xml

```diff
@@ -181,6 +181,14 @@
 			<argument index="1" name="placeholder" type="String" default="&quot;{_}&quot;" />
 			<description>
 				Formats the string by replacing all occurrences of [code]placeholder[/code] with [code]values[/code].
+				[b]Note:[/b] There must be one and only one [code]_[/code] in the [code]placeholder[/code].
+				[b]Note:[/b] The [code]values[/code] can be a dictionary, an one-dimensional array, or a two-dimensional array. When it's a two-dimensional array, the length of the inner arrays must be [code]2[/code], or an error will be generated.
```

Comment:
`Array` elements are not required to use the same type. So it does not have to be a strictly two-dimensional array.

```gdscript
# Prints: User 42 is Godot.
print("User {0} is {name}.".format([42, ["name", "Godot"]]))
```

Reply:
It seems quite complicated.
The elements in the outer array can be a mixture of a single `Variant` and a two-element array.
The elements in the inner array are also not specified must be of type `String`.